packages/client/src/DataCommonsClient.ts

```diff
@@ -0,0 +1,480 @@
+  /** Parent entity DCID. Example: `"country/USA"` */
+  parentEntity: string;
+  /** Child node type. Example: `"State"` */
+  childType: string;
+}
+
+export interface GetDataRowsParamsEntities extends BaseGetDataRowsParams {
+  /** Entity DCIDs. Example: `["country/USA", "country/IND"]` */
+  entities: string[];
+}
+export type GetDataRowsParams =
+  | GetDataRowsParamsWithin
+  | GetDataRowsParamsEntities;
+
+export type GetGeoJSONParams = GetDataRowsParams & {
+  /** GeoJSON property name in the knowledge graph. Inferred if not provided. */
+  geoJsonProperty?: string;
+};
+
+export interface DataRow {
+  [key: string]: string | number | boolean | null;
+}
+
+export type NodePropValues = Record<string, Record<string, string | null>>;
+
+class DataCommonsWebClient {
+  apiRoot?: string;
+
+  constructor(params?: DatacommonsWebClientParams) {
+    const p = params || {};
+    this.apiRoot =
+      p.apiRoot !== undefined
+        ? p.apiRoot.replace(/\/$/, "")
+        : "https://datacommons.org";
+  }
+
+  async getCsv(params: GetDataRowsParams): Promise<string> {
+    const dataRows = await this.getDataRows(params);
+
+    if (dataRows.length === 0) {
+      return "";
+    }
+
+    const header = Object.keys(dataRows[0]).sort();
+    const rows = dataRows.map((dataRow) =>
+      header.map((column) => dataRow[column])
+    );
+    const csvRows = [header, ...rows];
+    const csvLines = csvRows.map(encodeCsvRow);
+    return csvLines.join("\n");
+  }
+
+  async getGeoJSON(params: GetGeoJSONParams): Promise<FeatureCollection> {
+    const geoJsonProperty = params.geoJsonProperty || "geoJsonCoordinatesDP1";
+    const dataRows = await this.getDataRows({
+      ...params,
+      entityProps: [
+        geoJsonProperty,
+        ...(params.entityProps ? params.entityProps : DEFAULT_ENTITY_PROPS),
+      ],
+    });
+
+    const geoJson: FeatureCollection = {
+      type: "FeatureCollection",
+      features: dataRows
+        .filter((dataRow) => {
+          const geometryString = dataRow[`entity.${geoJsonProperty}`];
+          if (geometryString !== "string") {
+            return true;
+          }
+        })
+        .map((dataRow) => {
+          const geometryString = dataRow[`entity.${geoJsonProperty}`] as string;
+          const geometry = JSON.parse(geometryString) as Geometry;
+          const properties = { ...dataRow };
+          delete properties[`entity.${geoJsonProperty}`];
+          const feature: Feature = {
+            type: "Feature",
+            properties,
+            geometry,
+          };
+          return feature;
+        }),
+    };
+    return geoJson;
+  }
+
+  /**
+   * Fetches data commons variable values about an entity or entities.
+   * @param params
+   * @returns
+   */
+  async getDataRows(params: GetDataRowsParams): Promise<DataRow[]> {
+    // Fetch variable observations
+    const pointApiResponse =
+      "parentEntity" in params
```

Comment:
maybe should check `typeof params` instead because param fields could change and this could easily get missed?

Reply:
I tried a few ways to do this, and the `"parentEntity" in params` was the only thing that seemed to work. `typeof params` always returns the string `"object"`, and `instanceof` doesn't work for types, only classes 🤷 

https://www.turing.com/kb/check-type-of-objects-variables-in-typescript